src/client/auth.ts

```diff
@@ -914,3 +944,22 @@ export async function registerClient(
 
   return OAuthClientInformationFullSchema.parse(await response.json());
 }
+
+/**
+ * Internal helper to resolve DCR registration access token from provider and environment.
+ * Implements a clean 2-level fallback: provider method → environment variable.
+ */
+async function resolveDcrToken(provider?: OAuthClientProvider): Promise<string | undefined> {
```

Comment:
I don't think we should have the fallback, if you want to have the fallback to the environment variable in your Provider implementation, that's okay, but it doesn't need to be in the main flow.